url4fdf79ae2c_beckchr_juel_tgz-pJ8-de_odysseus_el_samples_xml_sax_XMELFilterJ8/src/javax/el/ListELResolver.java

```diff
@@ -0,0 +1,321 @@
+	 *            coerced into an integer, but otherwise ignored by this resolver.
+	 * @param value
+	 *            The value to be set at the given index.
+	 * @throws ClassCastException
+	 *             if the class of the specified element prevents it from being added to this list.
+	 * @throws PropertyNotFoundException
+	 *             if the given index is out of bounds for this list.
+	 * @throws PropertyNotWritableException
+	 *             if this resolver was constructed in read-only mode, or if the set operation is
+	 *             not supported by the underlying list.
+	 * @throws IllegalArgumentException
+	 *             if the property could not be coerced into an integer.
+	 * @throws NullPointerException
+	 *             if context is null
+	 * @throws ELException
+	 *             if an exception was thrown while performing the property or variable resolution.
+	 *             The thrown exception must be included as the cause property of this exception, if
+	 *             available.
+	 */
+	@Override
+	@SuppressWarnings("unchecked")
+	public void setValue(ELContext context, Object base, Object property, Object value) {
+		if (context == null) {
+			throw new NullPointerException("context is null");
+		}
+		if (isResolvable(base)) {
+			if (readOnly) {
+				throw new PropertyNotWritableException("resolver is read-only");
+			}
+			List list = (List) base;
+			int index = toIndex(list, property);
+			try {
+				list.set(index, value);
+			} catch (UnsupportedOperationException e) {
+				throw new PropertyNotWritableException(e);
+			} catch (ArrayStoreException e) {
+				throw new IllegalArgumentException(e);
+			}
+			context.setPropertyResolved(true);
+		}
+	}
+
+	/**
+	 * Test whether the given base should be resolved by this ELResolver.
+	 * 
+	 * @param base
+	 *            The bean to analyze.
+	 * @param property
+	 *            The name of the property to analyze. Will be coerced to a String.
+	 * @return base instanceof List
+	 */
+	private static final boolean isResolvable(Object base) {
+		return base instanceof List<?>;
+	}
+
+	/**
+	 * Convert the given property to an index in (list) base.
+	 * 
+	 * @param base
+	 *            The bean to analyze.
+	 * @param property
+	 *            The name of the property to analyze. Will be coerced to a String.
+	 * @return The index of property in base.
+	 * @throws IllegalArgumentException
+	 *             if base property cannot be coerced to an integer.
+	 * @throws PropertyNotFoundException
+	 *             if base is not null and the computed index is out of bounds for base.
+	 */
+	private static final int toIndex(List<?> base, Object property) {
+		int index = 0;
+		if (property instanceof Number) {
+			index = ((Number) property).intValue();
+		} else if (property instanceof String) {
+			try {
+				index = Integer.valueOf((String) property);
+			} catch (NumberFormatException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)